src/groovy/au/org/emii/aatams/detection/DetectionView.groovy

```diff
@@ -0,0 +1,177 @@
+    }
+
+    Project getTagProject() {
+        return Project.get(tagProjectId)
+    }
+
+    // Alias for transmitterProject - for compatibility with existing code - VisibilityControlService.
+    Project getProject() {
+        return this.tagProject
+    }
+
+    Surgery getSurgery() {
+        return Surgery.get(surgeryId)
+    }
+
+    String getSpeciesName() {
+        if (!spcode) {
+            return ''
+        }
+
+        return "${spcode} - ${scientificName} (${commonName})"
+    }
+
+    def applyEmbargo(allowSanitised = true) {
+
+        def releaseEmbargoed = surgery.release.isEmbargoed()
+
+        def censoredDetection = new HashMap(this.properties)
+
+        if (releaseEmbargoed) {
+            censoredDetection.surgery = null
+            censoredDetection.spcode = ''
+            censoredDetection.sensorId = ''
+            censoredDetection.speciesName = ''
+        }
+        else {
+            censoredDetection.surgery = surgery
+            censoredDetection.sensorId = sensorId
+            censoredDetection.speciesName = speciesName
+        }
+
+        censoredDetection.isSanitised = { -> true }
+
+        def protectionRequired = project.isProtected && releaseEmbargoed
+
+        def hideFromResults = releaseEmbargoed && !allowSanitised
+
+        if (protectionRequired || hideFromResults) {
+            return null
+        }
+
+        return censoredDetection
+    }
+
+    def isSanitised() {
+        !sensorId && !speciesName
+    }
+
+    static def fromSqlRow(def row, def detection) {
+        Detection.fromSqlRow(row, detection)
+
+        detection.commonName = row.common_name
+        detection.embargoDate = new DateTime(row.embargo_date).withZone(DateTimeZone.UTC)
+        detection.invalidReason = row.invalid_reason
+        detection.organisationName = row.organisation_name
+        detection.receiverDeploymentId = row.receiver_deployment_id
+        detection.releaseId = row.release_id
+        detection.releaseProjectId = row.release_project_id
+        detection.tagProjectId = row.tag_project_id
+        detection.scientificName = row.scientific_name
+        detection.sensorId = row.sensor_id
+        detection.spcode = row.spcode
+        detection.stationId = row.station_id
+        detection.stationLatitude = row.station_latitude
+        detection.stationLongitude = row.station_longitude
+        detection.stationStationName = row.station_station_name
+        detection.surgeryId = row.surgery_id
+        detection.uploader = row.uploader
+
+        return detection as DetectionView
+
+    }
+
+    static def fromSqlRow(def row) {
+        def detection = [:]
+        return fromSqlRow(row, detection)
+    }
+
+    // GORM-like methods.
+    def refresh(dataSource) {
+        get(this.id, dataSource, this)
+    }
+
+    static Detection get(id, dataSource) {
+        return get(id, dataSource, [:])
+    }
+
+    static Detection get(id, dataSource, detection) {
```

Comment:
If you wanted you could add a default value for detection thereby making the other get() method redundant.

```groovy
static Detection get(id, dataSource, detection = [:]) {
    ...
}
```